File: 06-Pila/cpp/main.cpp
#include "stackNode.h"
#include "stackArray.h"

using namespace std;

int main() {
    stackNode<int> *nodeStack = new stackNode<int>();

    cout << "Stack using nodes" << "\n";
    cout << "isEmpty(): " << nodeStack->isEmpty() << "\n";
    cout << "length(): " << nodeStack->getLength() << "\n";
    cout << "push(1)" << "\n";
    nodeStack->push(1);
    cout << "push(2)" << "\n";
    nodeStack->push(2);
    cout << "push(3)" << "\n";
    nodeStack->push(3);
    cout << *nodeStack << "\n";
    cout << "peek(): " << nodeStack->peek() << "\n";
    cout << "pop(): " << nodeStack->pop() << "\n";
    cout << *nodeStack << "\n";

    stackArray<int> *arrayStack = new stackArray<int>(5);
    cout << "Stack using Array" << "\n";
    cout << "isEmpty(): " << arrayStack->isEmpty() << "\n";
    cout << "length(): " << arrayStack->getLength() << "\n";
    cout << "push(1)" << "\n";
    arrayStack->push(1);
    cout << "push(2)" << "\n";
    arrayStack->push(2);
    cout << "push(3)" << "\n";
    arrayStack->push(3);
    cout << "push(4)" << "\n";
    arrayStack->push(4);
    cout << "push(5)" << "\n";
    arrayStack->push(5);
    cout << "isFull(): " << arrayStack->isFull() << "\n";
    cout << *arrayStack << "\n";
    cout << "peek(): " << arrayStack->peek() << "\n";
    cout << "pop(): " << arrayStack->pop() << "\n";
    cout << *arrayStack << "\n";

}

File: 06-Pila/python/main.py
from stackNode import StackNode
from stackArray import StackArray

nodeStack = StackNode();

print("Stack using nodes")
print("isEmpty(): " + str(nodeStack.isEmpty()))
print("length(): " + str(nodeStack.getLength()))
print("push(1)")
nodeStack.push(1)
print("push(2)")
nodeStack.push(2)
print("push(3)")
nodeStack.push(3)
print(nodeStack)
print("peek(): " + str(nodeStack.peek()))
print("pop(): " + str(nodeStack.pop()))
print(nodeStack)

print("")

arrayStack = StackArray(5)
print("Stack using Array")
print("isEmpty(): " + str(arrayStack.isEmpty()))
print("length(): " + str(arrayStack.getLength()))
print("push(1)")
arrayStack.push(1)
print("push(2)")
arrayStack.push(2)
print("push(3)")
arrayStack.push(3)
print("push(4)")
arrayStack.push(4)
print("push(5)")
arrayStack.push(5)
print("isFull(): " + str(arrayStack.isFull()))
print(arrayStack)
print("peek(): " + str(arrayStack.peek()))
print("pop(): " + str(arrayStack.pop()))
print(arrayStack)
